src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -66,7 +69,7 @@ public class TopQueriesService {
     /**
      * The internal thread-safe store that holds the top n queries insight data
      */
-    private final PriorityQueue<SearchQueryRecord> topQueriesStore;
+    private final PriorityQueueTopQueriesStore<SearchQueryRecord> priorityQueueTopQueriesStore;
```

Comment:
We can use interface now. PriorityQueueTopQueriesStore -> TopQueriesStore

And rename the variable to ```topQueriesStore```